components/cache.rst

```diff
@@ -23,112 +24,119 @@ Installation
+    The component also contains adapters to convert between PSR-6, PSR-16 and
+    Doctrine caches. See :doc:`/components/cache/psr6_psr16_adapters` and
+    :doc:`/components/cache/adapters/doctrine_adapter`.
 
-.. _cache-component-psr16-caching:
+.. _cache-component-contracts:
 
-Simple Caching (PSR-16)
------------------------
+Cache Contracts
+---------------
 
-This part of the component is an implementation of `PSR-16`_, which means that its
-basic API is the same as defined in the standard. First, create a cache object from
-one of the built-in cache classes. For example, to create a filesystem-based cache,
-instantiate :class:`Symfony\\Component\\Cache\\Simple\\FilesystemCache`::
+All adapters supports the  Cache Contract. It contains only two methods; ``get`` and
+``delete``. The first thing you need is to instantiate a cache adapter. The
+:class:`Symfony\\Component\\Cache\\Simple\\FilesystemCache` is used in this example::
 
-    use Symfony\Component\Cache\Simple\FilesystemCache;
+    use Symfony\Component\Cache\Adapter\FilesystemAdapter;
 
-    $cache = new FilesystemCache();
+    $cache = new FilesystemAdapter();
 
-Now you can create, retrieve, update and delete items using this object::
+Now you can retrieve and delete cached data using this object::
 
-    // save a new item in the cache
-    $cache->set('stats.products_count', 4711);
+    use Symfony\Contracts\Cache\ItemInterface;
 
-    // or set it with a custom ttl
-    // $cache->set('stats.products_count', 4711, 3600);
+    // The callable will only be executed on a cache miss.
+    $value = $cache->get('my_cache_key', function (ItemInterface $item) {
+        $item->expiresAfter(3600);
 
-    // retrieve the cache item
-    if (!$cache->has('stats.products_count')) {
-        // ... item does not exists in the cache
-    }
+        // ... do some HTTP request or heavy computations
+        $computedValue = 'foobar';
 
-    // retrieve the value stored by the item
-    $productsCount = $cache->get('stats.products_count');
+        return $computedValue;
+    });
+
+    echo $value; // 'foobar'
+
+    // ... and to remove the cache key
+    $cache->delete('my_cache_key');
 
-    // or specify a default value, if the key doesn't exist
-    // $productsCount = $cache->get('stats.products_count', 100);
+.. note::
 
-    // remove the cache key
-    $cache->delete('stats.products_count');
+    Use tags to clear more than one key at the time. Read more at
+    :doc:`/components/cache/cache_invalidation`.
 
-    // clear *all* cache keys
-    $cache->clear();
+The Cache Contracts also comes with built in `Stampede prevention`_. This will
+remove CPU spikes at the moments when the cache is cold. If an example application
+spends 5 seconds to compute data that is cached for 1 hour. This data is accessed
+10 times every second. This means that you mostly have cache hits and everything
+is fine. But after one hour, we get 10 new requests to a cold cache. So we start
```

Comment:
Always try to avoid first person:

```rst
is fine. But after one hour, you get 10 new requests to a cold cache. So the data
is computed again. The next second the same thing happens. So the data is computed
about 50 times before the cache is warm again. This is where you need stampede
prevention
```